### PR TITLE
(chore) MARXN23-6 improve test isolation

### PR DESCRIPTION
**Pull request by alexeh**
Superseeds [#935](https://github.com/Vizzuality/marxan-cloud/pull/935)

### Overview

From [Jira issue](https://vizzuality.atlassian.net/browse/MRXN23-3):

This PR aims to remove dependency of seeding the database via Makefile.
Seeding and Cleanup / TearDown will be handled in a single point within the application.

Plus adds new env configuration to make possible for devs to run tests locally

### Testing instructions

-CI should be all green
-Devs should be able to run tests natively

### Feature relevant tickets

[Jira issue](https://vizzuality.atlassian.net/browse/MRXN23-3)

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

**Comment by hotzevzl** [comment]
@alexeh besides what we discussed today, not much else from my end - given the amount of changes in `develop` between the original PR and now, I'd consider it mostly acceptable to bring in `theirs` changes when rebasing (which is what you did as far as you could), focus on getting the services building, starting and running (we can run bots against them, once I finish my updates on seeds) and eventually tests pass again, less the (hopefully few) ones we consider ok to skip temporarily.

File: api/apps/api/config/default.json [hunk 1]
```diff
@@ -58,6 +58,11 @@
   "webshot": {
     "url": "http://webshot:3000"
   },
+
+  "cloning": {
```
Comment (alexeh) [inline api/apps/api/config/default.json:62]: Add just in test

File: api/apps/api/test/utils/handle-database.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  geoConnection = await createConnection(apiConnections.geoprocessingDB);
+
+  console.log(apiConnection.isConnected, geoConnection.isConnected);
+
+  await ensureThereAreNotPendingMigrations(apiConnection, 'API');
+  await ensureThereAreNotPendingMigrations(geoConnection, 'Geoprocessing');
+});
+
+beforeEach(async () => {
+  await clearTables(apiConnection, ['roles', 'api_event_kinds']);
```
Comment (alexeh) [inline api/apps/api/test/utils/handle-database.ts:20]: Fix this

File: api/apps/api/test/utils/handle-database.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  await clearTables(geoConnection);
+  await insertAdminRegions(geoConnection);
```
Comment (alexeh) [inline api/apps/api/test/utils/handle-database.ts:22]: Check where is already needed, or insert just once as beforeAll global hook

File: api/package.json [hunk 4]
```diff
@@ -22,11 +22,14 @@
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "api:test:e2e": "NODE_OPTIONS=\"--max_old_space_size=3072\" jest --config ./apps/api/test/jest-e2e.json",
-    "geoprocessing:test:e2e": "jest --config ./apps/geoprocessing/test/jest-e2e.json",
+    "api:test:e2e": "NODE_OPTIONS=\\\"--max_old_space_size=3072\\\" jest --runInBand --forceExit --ci --config apps/api/test/jest-e2e.json",
+    "geoprocessing:test:e2e": "jest --runInBand --ci --detectOpenHandles --forceExit --config ./apps/geoprocessing/test/jest-e2e.json",
```
Comment (alexeh) [inline api/package.json:26]: add migration run here

File: api/apps/api/test/utils/handle-database.ts (hunk 2 above)
Comment (alexeh) [reply api/apps/api/test/utils/handle-database.ts:20]: TODO

**Pull request merged**